diff --git a/include/configs/mx6cuboxi.h b/include/configs/mx6cuboxi.h
index 446b3bc..8d87400 100644
--- a/include/configs/mx6cuboxi.h
+++ b/include/configs/mx6cuboxi.h
@@ -143,8 +143,15 @@
        (CONFIG_SYS_INIT_RAM_ADDR + CONFIG_SYS_INIT_SP_OFFSET)
 
 /* Environment organization */
-#define CONFIG_ENV_SIZE                        (8 * 1024)
-#define CONFIG_ENV_IS_IN_MMC
-#define CONFIG_ENV_OFFSET              (8 * 64 * 1024)
+#undef  CONFIG_ENV_IS_NOWHERE
+#undef  CONFIG_ENV_IS_IN_NAND
+#undef  CONFIG_ENV_IS_IN_MMC
+#define CONFIG_FAT_WRITE
+#define CONFIG_ENV_IS_IN_FAT
+#define CONFIG_ENV_SIZE                        (128 * 1024)
+#define FAT_ENV_DEVICE_AND_PART         "0:1"
+#define FAT_ENV_INTERFACE               "mmc"
+#define FAT_ENV_FILE                    "uboot.env"
+#define CONFIG_SYS_REDUNDAND_ENVIRONMENT // After discussion on IRC we use 5 byte headers to make life easier.
 
 #endif                         /* __MX6CUBOXI_CONFIG_H */

